packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -80,3 +93,672 @@ class CupertinoColors {
+      final bool isHighContrastEnabled = MediaQuery.of(context, nullOk: nullOk)?.highContrast
+        ?? false;
+
+        highContrastNumber = isHighContrastEnabled ? 1 : 0;
+    }
+
+    // If this CupertinoDynamicColor cares about user interface elevation.
+    if (_isInterfaceElevationDependent) {
+      final CupertinoUserInterfaceLevelData level =
+        CupertinoUserInterfaceLevel.of(context, nullOk: nullOk)
+        ?? CupertinoUserInterfaceLevelData.base;
+
+      interfaceElevationNumber = level.index;
+    }
+
+    final Color resolved = _colorMap[brightnessNumber][highContrastNumber][interfaceElevationNumber];
+    return resolved.value == value ? this : CupertinoDynamicColor._(resolved, _colorMap);
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    return other.runtimeType == runtimeType
+        && value == other.value
+        && const ListEquality<List<List<Color>>>(ListEquality<List<Color>>(ListEquality<Color>())).equals(_colorMap, other._colorMap);
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      value,
+      color,
+      darkColor,
+      highContrastColor,
+      elevatedColor,
+      darkElevatedColor,
+      darkHighContrastColor,
+      darkHighContrastElevatedColor,
+      highContrastElevatedColor,
+    );
+  }
+
+  @override
+  String toString() {
+    String toString(String name, Color color) {
+      final String marker = color.value == value ? '*' : '';
+      return '$marker$name = $color$marker';
+    }
+
+    final List<String> xs = <String>[
+      toString('color', color),
+      if (_isPlatformBrightnessDependent) toString('darkColor', darkColor),
+      if (_isHighContrastDependent) toString('highContrastColor', highContrastColor),
+      if (_isPlatformBrightnessDependent && _isHighContrastDependent) toString('darkHighContrastColor', darkHighContrastColor),
+      if (_isInterfaceElevationDependent) toString('elevatedColor', elevatedColor),
+      if (_isPlatformBrightnessDependent && _isInterfaceElevationDependent) toString('darkElevatedColor', darkElevatedColor),
+      if (_isHighContrastDependent && _isInterfaceElevationDependent) toString('highContrastElevatedColor', highContrastElevatedColor),
+      if (_isPlatformBrightnessDependent && _isHighContrastDependent && _isInterfaceElevationDependent) toString('darkHighContrastElevatedColor', darkHighContrastElevatedColor),
+    ];
+
+    return '$runtimeType(${xs.join(', ')})';
+  }
+}
+
+/// A color palette that typically matches iOS 13+ system colors.
+///
+/// Generally you should not create a [CupertinoSystemColorsData] yourself.
+/// Use [CupertinoSystemColors.of] to get the [CupertinoSystemColorsData] from the
+/// current [BuildContext] if possible, or [CupertinoSystemColors.fromSystem]
+/// when the current [BuildContext] is not available (e.g., in [CupertinoApp]'s
+/// constructor).
+@immutable
+class CupertinoSystemColorsData extends Diagnosticable {
```

Comment:
This class should have lerp and debugFillProperties methods, see packages/flutter/lib/src/material/tooltip_theme.dart for examples.

There are some boilerplate-style tests to write for FooData classes like this. See the first 3 tests in packages/flutter/tests/material/tooltip_theme_tests.dart for examples.

Reply:
What would be the use case of lerping `CupertinoSystemColorsData`?

Reply:
Not sure it applies in this case, but in other theme data classes they are usually a field in the overall ThemeData which needs to lerp all of its fields in its own lerp function (which is used for animating theme changes).

Reply:
I can't really think of a use case where lerping the entire `CupertinoSystemColorsData`, and on iOS light/dark change doesn't seem to animate